spec/did.md

```diff
@@ -0,0 +1,80 @@
+
+## DID Document Data Model
+
+Following from [this data model](https://www.w3.org/TR/did-core/#core-properties).
+
+| Property      | JSON Representation | Required | Notes          |
+| ------------- | ------------------- | -------- | -------------- |
+| `id`          | String              | Yes      | Must be a URI. |
+| `@context`    | Array of strings    | No       | Depends on the DID method. |
+| `controller`  | Array of strings    | No       | Depends on the DID method. Strings must be URIs. |
+| `alsoKnownAs`        | Array of strings | No   | Depends on the DID method. Strings must be URIs. |
+| `verificationMethod` | Array of [Verification Methods](#verification-method-data-model) | Yes | There must be at least one Verification Method in each DID Document. |
+| `authentication`     | Array of strings | No   | String values are recommended to be fully qualified DID URIs (e.g. `did:example:abcd#key-1` over `#key-1`). |
+| `assertion`          | Array of strings | No   | String values are recommended to be fully qualified DID URIs (e.g. `did:example:abcd#key-1` over `#key-1`). |
+| `keyAgreement`       | Array of strings | No   | String values are recommended to be fully qualified DID URIs (e.g. `did:example:abcd#key-1` over `#key-1`). |
+| `capabilityInvocation` | Array of strings | No | String values are recommended to be fully qualified DID URIs (e.g. `did:example:abcd#key-1` over `#key-1`). |
+| `capabilityDelegation` | Array of strings | No | String values are recommended to be fully qualified DID URIs (e.g. `did:example:abcd#key-1` over `#key-1`). |
+| `service`            | Array of [Services](#service-data-model) | No | - |
+
+**Additional Notes:**
+- No [JSON-LD processing](https://www.w3.org/TR/did-core/#consumption-0) is performed.
+- Each [Verification Method](https://www.w3.org/TR/did-core/#verification-methods) must have at least one [Verification Relationship](https://www.w3.org/TR/did-core/#verification-relationships).
+
+### Verification Method Data Model
+
+Following form [this data model](https://www.w3.org/TR/did-core/#verification-methods).
+
+| Property      | JSON Representation | Required | Notes          |
+| ------------- | ------------------- | -------- | -------------- |
+| `id`          | String              | Yes      | Recommended to be a fully qualified DID URI (e.g. `did:example:abcd#key-1`). |
+| `type`        | String              | Yes      | Must be a URI. |
+| `controller`  | String              | Yes      | Must be a URI. |
+| `publicKeyJwk`        | Object      | No       | Preferred. Represents a [JWK](https://www.w3.org/TR/did-core/#bib-rfc7517). Either this or `publicKeyMultibase` must be present. |
```

Comment:
at least in kotlin right now, we throw an exception if publicKeyJwk is missing while building VerificationMethod...

Reply:
that's probably fine. I was afraid of strengthening this language since I know the existing did key impls use multibase